docs/architecture/transaction.md

```diff
@@ -0,0 +1,64 @@
+# Transaction
+
+> Single account state transition
+
+## What is the purpose of a transaction?
+
+To facilitate single account state transitions.
+
+## What is a transaction?
```

Comment:
I'm not sure we need to so many sub sections - I'd probably just have a single title `Transactions` and then a couple of short paragraphs before going into specifics. Maybe I'd start with something like:

"Transactions are a mechanism by which the state of the chain is updated. That is, all changes in account and note states result from executing transactions. In Miden, transactions can originate either from the users or from the network itself ..."

I would also try to weave in design rationale as we explain things to make the reading more interesting (rather than  putting rationale into a separate section). For example, the main goal of this design to make all transactions independently executable/provable.